### ISSUE TITLE
[request] support for vendor-defined HID devices

### ISSUE THREAD

**Issue by Kanuan** [Ryochan7/DS4Windows#2045]
This is related to Nefarius' DsHidMini.

Because DS3s using DsHidMini in DS4 Mode are presented to the system as a gamepad, HidGuardian/HidHide is needed to solve double input issues. Having the controller be presented as something else to the system would prevent the issue, and not needing HidGuardian/HidHide means less apps that the user needs to install and setup for this use case.

The idea is to modify DsHidMini's DS4 Mode HID Descriptor to present the device as a vendor-defined HID Device instead of a Gamepad, but making this modification requires changes on DS4Windows side since it doesn't recognize the controller anymore.

[This branch](https://github.com/Kanuan/DsHidMini/tree/DS4Vendor) of DsHidMini has the HID Descriptor change in place and [this file is the compiled modified driver](https://github.com/Ryochan7/DS4Windows/files/6284915/DsHidMini.Test.Build.-.DS4.mode.as.Vendor-Defined.zip) (requires windows being in test mode)

I appreciate your insights on if this is possible without big changes on DS4Windows. I tried looking at DS4Windows' code but I only know the basics of programming and this is beyond my skill level

**Comment by Ryochan7** [comment]
The major problem would be that DS4Windows does look for the Usage ID when enumerating devices. It looks for 0x05 or 0x04 in order to identify a proper gamepad device from some other exposed devices for some controllers. Some special exception would have to be made for this particular case in order to have DS4Windows return a reference to that HID device.

https://github.com/Ryochan7/DS4Windows/blob/jay/DS4Windows/HidLibrary/HidDevices.cs#L64

**Comment by Kanuan** [comment]
I just changed the usage from 0x01 to 0x05 and now DS4Windows is detecting the controller, but since the Usage page is Vendor defined windows is ignoring the main controller.

This was so easy that I'm having trouble believing it. Am I doing something wrong? Nope, hidhide is disabled, can't be it. 

I will do some more tests to believe in what is in front of me, but everything seems fine

**Comment by Kanuan** [comment]
@Ryochan7 [I added Generic Desktop Controls detection to this branch](https://github.com/Ryochan7/DS4Windows/compare/jay...Kanuan:HID_USAGE_GENERIC) so I could run some test. Since DS4Windows compares VID/PID on the supported device list I'm supposing this change won't cause issues with DS4W picking devices it shouldn't, but I'd like your thoughts on this.

The first approach I tried in defining the DS3 controller as a `Vendor defined` device but with a `gamepad` usage, so ds4windows could pick the controller without changes, is flawed because even though windows ignores the controller, some aplications (e.g. [Gamepad Tester wensite](https://gamepad-tester.com/)) can still pick it (even if no input actually detected). We would like to prevent any chance of the controller being detected by mistake.

Setting the device as `Vendor defined` with `Generic desktop controls` usage seems to make the controller be fully ignored by games and applications. I'm making tests to see if this approach doesn't cause some obscure issue, but if this works fine then the new detection change would need to be implemented on DS4W.
